### ISSUE TITLE
Error creating teams with token

### ISSUE THREAD

**Issue by ichasco** [grafana/terraform-provider-grafana#146]
Hi there,

I am getting problems when creating resources with tokens. Terraform gives me a error but it works (I don't know why)

### Terraform Version
```
Terraform v0.13.4
+ provider registry.terraform.io/grafana/grafana v1.7.0
+ provider registry.terraform.io/hashicorp/vault v2.16.0
```

### Affected Resource(s)

- grafana_team

### Terraform Configuration Files
Provider
```hcl

## ADMIN

data "vault_generic_secret" "grafana_admin" {
  path = "devops/Grafana/Users/Admin"
}

provider "grafana" {
  alias = "admin"
  url  = "https://grafana.example.com"
  auth = "admin:${data.vault_generic_secret.grafana_admin.data["Password"]}"

}

## MAIN ORG

data "vault_generic_secret" "grafana_token" {
  path = "devops/Grafana/Tokens"
}

provider "grafana" {
  url  = "https://grafana.example.com"
  auth = data.vault_generic_secret.grafana_token.data["Token-Terraform"]
}

```

Create ORG

```hcl
resource "grafana_organization" "mainorg" {
  provider    = grafana.admin
  name        = "MainOrg"
  admin_user  = "admin"
}
```
```
terraform apply -target=grafana_organization.mainorg
```

Create the token manually in created org with Admin permission

Create Team
```hcl
resource "grafana_team" "devops" {
  name    = "devops"
}

resource "grafana_team_preferences" "team_preferences" {
  team_id           = grafana_team.devops.id
  theme             = "dark"
  timezone          = "browser"
}
```

```
terraform apply
```

### Debug Output

Create ORG output
```
data.vault_approle_auth_backend_role_id.role: Refreshing state... [id=auth/approle/role/grafana-devops/role-id]
vault_approle_auth_backend_role_secret_id.id: Refreshing state... [id=backend=approle::role=grafana-devops::accessor=XXXXXXXXXXX]
vault_approle_auth_backend_login.login: Refreshing state... [id=XXXXXXXXXXXXXXXX]
data.vault_generic_secret.grafana_admin: Refreshing state... [id=devops/Grafana/Users/Admin]
data.vault_generic_secret.grafana_token: Refreshing state...
grafana_organization.mainorg: Refreshing state... [id=2]
```

Create Team output

```
grafana_team.devops: Creating...
Error: status: 403, body: {"message":"Permission denied"}

  on teams.tf line 1, in resource "grafana_team" "devops":
   1: resource "grafana_team" "devops" {
```

Grafana Logs

```
grafana         | t=2020-12-07T14:18:04+0100 lvl=eror msg="Team not found" logger=context userId=0 orgId=3 uname= error="Team not found" remote_addr=3.138.134.63
grafana         | t=2020-12-07T14:18:04+0100 lvl=info msg="Request Completed" logger=context userId=0 orgId=3 uname= method=GET path=/api/teams/1 status=404 remote_addr=3.138.134.63 time_ms=18 size=28 referer=
grafana         | t=2020-12-07T14:18:40+0100 lvl=info msg="Request Completed" logger=context userId=0 orgId=3 uname= method=GET path=/api/users status=403 remote_addr=35.175.216.88 time_ms=29 size=31 referer=
```

But the team has been created:

```
curl -H "Authorization: Bearer XXXXXXXXXXXXXXXXXXXx" "https://grafana.example.com/api/teams/search?perpage=50&page=1"
{"totalCount":2,"teams":[{"id":2,"orgId":3,"name":"devops","email":"","avatarUrl":"/avatar/0e33bbb89785581715c04243c2624256","memberCount":0,"permission":0}],"page":1,"perPage":50}%
```

**Comment by benli0822** [comment]
Hi,

I think 

`resource "grafana_team" "devops" {
  name    = "devops"
}`

should be

`resource "grafana_team" "devops" {
  provider    = grafana.admin
  name    = "devops"
}`

in your case? Token API is a little bit ambiguous from the description on Grafana Team API BTW, https://grafana.com/docs/grafana/latest/http_api/team/

**Comment by jmvizcainoio** [comment]
Hi, 

i have the same problem. 


```
provider "grafana" {
  url    = "https://grafana_url/"
  auth   = "token"
  org_id = 1
}

resource "grafana_team" "business" {
  name = "business"
  members = ["email@domain.com"]
}
```

and the response is

```
grafana_team.business: Creating...

Error: status: 403, body: {"message":"Permission denied"}

  on main.tf line 7, in resource "grafana_team" "business":
   7: resource "grafana_team" "business" {

```

any update?

**Comment by mahsoud** [comment]
Facing the same issue with grafana/grafana v1.11.0 & terraform v0.15.0

**Comment by brunowego** [comment]
Use basic auth instead of token.

**Comment by trotttrotttrott** [comment]
https://github.com/grafana/terraform-provider-grafana/pull/244 should have resolved this. It was released in v1.13.2.
